harness/determined/core/_train.py

```diff
@@ -67,41 +67,60 @@ def _get_last_validation(self) -> Optional[int]:
         logger.debug(f"_get_last_validation() -> {steps_completed}")
         return steps_completed
 
-    def report_training_metrics(
+    def report_trial_metrics(
         self,
-        steps_completed: int,
+        metric_type: str,
+        total_batches: int,
         metrics: Dict[str, Any],
         batch_metrics: Optional[List[Dict[str, Any]]] = None,
     ) -> None:
         """
-        Report training metrics to the master.
+        Report trial metrics to the master.
 
         You can include a list of ``batch_metrics``.  Batch metrics are not be shown in the WebUI
         but may be accessed from the master using the CLI for post-processing.
         """
 
-        body = {
-            "trial_run_id": self._run_id,
-            "steps_completed": steps_completed,
-            "metrics": {
-                "avg_metrics": metrics,
-            },
-        }
-        if batch_metrics is not None:
-            body["metrics"]["batch_metrics"] = batch_metrics  # type: ignore
-        logger.info(
-            f"report_training_metrics(steps_completed={steps_completed}, metrics={metrics})"
-        )
-        self._session.post(
-            f"/api/v1/trials/{self._trial_id}/training_metrics",
-            data=det.util.json_encode(body),
+        # CHECK: do we want this for training as well? same for batch_metrics?
+        serializable_metrics = self._get_serializable_metrics(metrics)
+        reportable_metrics = {k: metrics[k] for k in serializable_metrics}
+
+        v1metrics = bindings.v1Metrics(avgMetrics=reportable_metrics, batchMetrics=batch_metrics)
+        v1TrialMetrics = bindings.v1TrialMetrics(
+            metrics=v1metrics,
+            stepsCompleted=total_batches,
+            trialId=self._trial_id,
+            trialRunId=self._run_id,
         )
+        body = bindings.v1ReportTrialMetricsRequest(metrics=v1TrialMetrics, type=metric_type)
+        bindings.post_ReportTrialMetrics(self._session, body=body, metrics_trialId=self._trial_id)
 
+        # Also sync tensorboard (all metrics, not just json-serializable ones).
         if self._tensorboard_mode == TensorboardMode.AUTO:
             if self._tbd_writer:
-                self._tbd_writer.on_train_step_end(steps_completed, metrics, batch_metrics)
+                if metric_type == util._LEGACY_VALIDATION:
+                    self._tbd_writer.on_validation_step_end(total_batches, metrics)
+                elif metric_type == util._LEGACY_TRAINING:
+                    self._tbd_writer.on_train_step_end(total_batches, metrics, batch_metrics)
+                else:
+                    pass  # FIXME
```

Comment:
what do you think should happen here? this was a non-existent branch before generic metrics.

Reply:
add a new method for tsb?
https://github.com/determined-ai/determined/blob/e23b58becdfe03a52fca01f031dce13e1287b340/harness/determined/tensorboard/metric_writers/callback.py#L21

Reply:
leaving it for a different branch/pr https://github.com/hamidzr/determined/tree/tsb-generic-metrics